### PR TITLE
[MM-58560] Configurable session revocation during password resets

### PR DESCRIPTION
**Pull request by DSchalla**
#### Summary
Currently Mattermost does not provide the option to perform session revocation upon password changes for end-users, which was an intentional design decision for UX reasons at some time ago. As part of this change, we will add a new option to toggle this behavior during password changes:

**Matrix:**
- If config is on and user session A changes password, all other but that sessions are revoked
- If config is off and user session A changes password, all other sessions remain
- If config is on and admin session changes password, all sessions for user are revoked
- If config is off and admin session changes password, all sessions remain

**Default:**
- Default Off for current installations
- Default On for new installations

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58560

#### Screenshots
No UI changes with the exception of a new boolean config in the Session Lengths config.

#### Release Note
```release-note
Added a new config setting ServiceSettings.TerminateSessionsOnPasswordChange to configure sessions revocation during password resets.
```

### CODE REVIEW

**Comment by DSchalla** [comment]
@cwarnermm This one will definitely need to be added in the Docs, I'll prepare a PR for that on Friday - Just a heads up.

**Review by wiersgallak — APPROVED** [review]
Thanks!

File: webapp/channels/src/components/admin_console/session_length_settings.tsx [hunk 1]
```diff
@@ -39,6 +40,8 @@ const messages = defineMessages({
...
     extendSessionLengthActivity_label: {id: 'admin.service.extendSessionLengthActivity.label', defaultMessage: 'Extend session length with activity: '},
     extendSessionLengthActivity_helpText: {id: 'admin.service.extendSessionLengthActivity.helpText', defaultMessage: 'When true, sessions will be automatically extended when the user is active in their Mattermost client. Users sessions will only expire if they are not active in their Mattermost client for the entire duration of the session lengths defined in the fields below. When false, sessions will not extend with activity in Mattermost. User sessions will immediately expire at the end of the session length or idle timeouts defined below. '},
+    terminateSessionsOnPasswordChange_label: {id: 'admin.service.terminateSessionsOnPasswordChange.label', defaultMessage: 'Terminate Sessions on Password Change: '},
+    terminateSessionsOnPasswordChange_helpText: {id: 'admin.service.terminateSessionsOnPasswordChange.helpText', defaultMessage: 'When true, all sessions of a user will expire if their password is changed by themselves or an administrator.'},
```
Comment (JulienTant) [inline webapp/channels/src/components/admin_console/session_length_settings.tsx:44]: nit: Not sure it's worth changing but for the user changing their password, it's all the sessions except for the current one.

**Review by JulienTant — APPROVED** [review]
Thanks @DSchalla !

**Review by esarafianou — APPROVED** [review]

**Review by mgdelacroix — APPROVED** [review]

File: webapp/channels/src/components/admin_console/session_length_settings.tsx (hunk 1 above)
Comment (DSchalla) [reply webapp/channels/src/components/admin_console/session_length_settings.tsx:44]: Thanks @JulienTant, I agree that would be good to add. I'll add some verbage.

**Pull request merged**